### PR TITLE
Backporting IUser/Collaboration PRs

### PR DESCRIPTION
**Pull request by martinRenou**
## References

Backports of all PRs concerning real-time collaboration UI changes (left panel, top menu) #11443 #11605 #11703 #11774 #12046 #12175 #12548 #12095

### CODE REVIEW

**Comment by jtpio** [comment]
Probably need https://github.com/jupyterlab/jupyterlab/pull/12586 or related to align the user menu to the right. Although this might be breaking for some folks / extensions already using the top area in 3.x.

**Comment by martinRenou** [comment]
You're commenting at the right time, I was looking for this! Thanks.

Do you think there is a way to do this in a non-breaking way?

**Comment by jtpio** [comment]
Probably the only "breaking" change would be visually with the addition of the spacer widget in the top area.

This might affect extensions that add widgets to the top area, similar to https://github.com/jtpio/jupyterlab-system-monitor.

Not sure we should do anything clever to handle this, or just document this change in a `3.5.x` -> `3.6.x` migration (similar to https://jupyterlab.readthedocs.io/en/latest/extension/extension_migration.html#jupyterlab-3-0-to-3-1).

Another option would be to leave the user menu indicator on the left hand side as is and not backport https://github.com/jupyterlab/jupyterlab/pull/12586.

**Comment by martinRenou** [comment]
CI failing with `error Couldn't find any versions for "@jupyterlab/collaboration" that matches "~3.5.0-beta.0"` as expected because it's not available yet.

File: tsconfigdoc.json [hunk 1]
```diff
@@ -280,6 +286,9 @@
     {
       "path": "./packages/ui-components-extension"
     },
+    {
+      "path": "./packages/user-extension"
+    },
```
Comment (fcollonval) [inline tsconfigdoc.json:291]: False positive integrity change

```suggestion
```

File: packages/application-extension/schema/top-bar.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
+{
+  "title": "Top Bar",
+  "description": "Top Bar settings.",
+  "jupyter.lab.toolbars": {
+    "TopBar": [
+      {
+        "name": "spacer",
+        "type": "spacer",
+        "rank": 50
+      }
```
Comment (fcollonval) [inline packages/application-extension/schema/top-bar.json:10]: I would not add the spacer here. And prefer to register a new widget in the collaboration-extension to the toolbar that create a spacer to be added with the user menu only in collaborative mode. This should help not impacting the UI in non-collaborative mode.

```suggestion
```

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @martinRenou 

I have some suggestion

File: packages/application-extension/schema/top-bar.json (hunk 2 above)
Comment (jtpio) [reply packages/application-extension/schema/top-bar.json:10]: > only in collaborative mode

Right that would be a good compromise.

**Comment by jtpio** [comment]
> CI failing with `error Couldn't find any versions for "@jupyterlab/collaboration" that matches "~3.5.0-beta.0"` as expected because it's not available yet.

Probably we can publish the package from this branch as is to make the CI pass.

**Comment by jtpio** [comment]
Just published `@jupyterlab/collaboration` and `@jupyterlab/collaboration-extension` to npm to help make the CI pass:

- https://www.npmjs.com/package/@jupyterlab/collaboration
- https://www.npmjs.com/package/@jupyterlab/collaboration-extension

**Comment by jtpio** [comment]
Thanks Marin for doing all these backports!

**Pull request merged**